### PR TITLE
Create Book model class

### PR DESCRIPTION
**Pull request by Dimagaa**
Create BookRepository interface and its implementation BookRepositoryImpl Create BookService interface and its implementation BookServiceImpl

### CODE REVIEW

File: src/test/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+spring.datasource.url=jdbc:h2:mem:testdb
```
Comment (andrii-hoienko) [inline src/test/resources/application.properties:1]: Are test properties necessary now?

File: src/main/java/com/app/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't persist new book: " + book, e);
```
Comment (andrii-hoienko) [inline src/main/java/com/app/onlinebookstore/repository/impl/BookRepositoryImpl.java]: Please use custom exception

File: src/main/java/com/app/onlinebookstore/OnlineBookStoreApplication.java [hunk 3]
```diff
@@ -1,12 +1,71 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class OnlineBookStoreApplication {
+    @Autowired
+    private BookService bookService;
```
Comment (andrii-hoienko) [inline src/main/java/com/app/onlinebookstore/OnlineBookStoreApplication.java]: It is better to use dependency injection via constructor

File: src/main/java/com/app/onlinebookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
```
Comment (andrii-hoienko) [inline src/main/java/com/app/onlinebookstore/model/Book.java:23]: isbn? what is that?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good, but let`s improve your solution

File: src/test/resources/application.properties (hunk 1 above)
Comment (Dimagaa) [reply src/test/resources/application.properties:1]: CI checks do not pass without that and this step is  included in the requirements

File: src/main/java/com/app/onlinebookstore/model/Book.java (hunk 4 above)
Comment (Dimagaa) [reply src/main/java/com/app/onlinebookstore/model/Book.java:23]: According to the requirements:

image

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**